spec/features/proposals/index_spec.rb

```diff
@@ -20,20 +20,72 @@
     user = create(:user)
     proposals = create_list(:proposal, 2, observer: user)
     cancelled = create_list(:proposal, 2, status: "cancelled", observer: user)
+    @page = MyRequestsPage.new
 
     login_as(user)
-    visit proposals_path
+    @page.load
+
+    expect(@page.pending.desc_column_header).to have_content "Submitted"
+    expect(@page.cancelled.desc_column_header).to have_content "Submitted"
+
+    expect_order(@page.pending, proposals.reverse)
+    expect_order(@page.cancelled, cancelled.reverse)
+  end
+
+  describe "status field text" do
+    context "when the user is an approver or purchaser" do
+      it "is correct for the user" do
+        user = create(:user)
+        other_user = create(:user)
+        approval_proposal = create_proposal_with_approvers(user, other_user)
+        purchase_proposal = create_proposal_with_approvers(other_user, user)
+        purchase_proposal.individual_steps.first.approve!
+        @page = MyRequestsPage.new
 
-    within(reviewable_proposals_section) do
-      expect(find("th.desc")).to have_content "Submitted"
+        login_as(user)
+        @page.load
+
+        expect(@page.needing_review.requests[1].public_id_link.text).to eq approval_proposal.public_id
+        expect(@page.needing_review.requests[1].status.text).to eq "Please review"
+        expect(@page.needing_review.requests[0].public_id_link.text).to eq purchase_proposal.public_id
+        expect(@page.needing_review.requests[0].status.text).to eq "Please purchase"
+      end
     end
 
-    expect_order(reviewable_proposals_section, proposals.reverse)
+    context "when the user's request is waiting for approval or purchase" do
+      it "is correct for the user" do
```

Comment:
same comment as above: what about a separate spec for each step type? More lines of code overall because you're doing double setup, but also easier to understand since each spec would only need setup for that step type rather than both (connection between setup and expectations would be clearer)